### ISSUE TITLE
[Question] Is it possible to return configuration of generator node 

### ISSUE THREAD

**Issue by LukaszDuss** [thingsboard/thingsboard#8314]
**Component**
 * Rule Engine
 
**Environment**
 * ThingsBoard: version 3.4.0 PE

**TL;DR**
I'd like to pass settings from generator node to next node in my rulechain, to be precise I need to return "Period in Seconds" value to next node. 
Is there a way to grab settings values/object in script box?

**Description**
I'm creating rulechain to conditionaly and periodically send email about inactive/disconnected devices. 
Through generator node im generating message with required originator that stores data about inactive devices.
In the next step id like to change hardcoded value to one that is obtained from passed generator setting value.

Generator node from exported rulechain JSON:
`{
        "additionalInfo": {
          "description": "",
          "layoutX": 853,
          "layoutY": 57
        },
        "type": "org.thingsboard.rule.engine.debug.TbMsgGeneratorNode",
        "name": "Time Interval",
        "debugMode": true,
        "configuration": {
          "msgCount": 0,
          "periodInSeconds": 300,
          "jsScript": "// ",
          "originatorId": "77a398a0-ce02-11ed-a4a1-d73c8b1a93a3",
          "originatorType": "DEVICE"
        },
        "externalId": null
      },`
Right now im not returning anything from generator script.

**Question**
How can I return configuration.periodInSecond, or rather how to obtain configuration object within sctipt?

P.S. I'm aware that I should use scheduler instead of generator. I will check if i can recreate my functionality with scheduler in future

**Comment by LukaszDuss** [comment]
Or maybe other way around - can i set configuration/settings field from script box?

**Comment by ssakharova** [comment]
Hello @LukaszDuss,

You are correct that you should use a scheduler instead of a generator rule node for your current needs.
